src/components/SignInCard.tsx

```diff
@@ -0,0 +1,189 @@
+import { CONNECTION_STATE } from '@/context/constants'
+import { useWeb3Context } from '@/context/useWeb3Context'
+import { signUpWithCircle } from '@/server/actions'
+import { getCsrfToken, signIn, useSession, signOut } from 'next-auth/react'
+import { SiweMessage } from 'siwe'
+import {
+    useAccount,
+    useConnect,
+    useNetwork,
+    useSignMessage,
+    useDisconnect,
+} from 'wagmi'
+import { InjectedConnector } from 'wagmi/connectors/injected'
+import {
+    Card,
+    CardHeader,
+    Flex,
+    Heading,
+    CardBody,
+    Button,
+    Box,
+    Stack,
+    Center,
+    useDisclosure,
+    Modal,
+    ModalOverlay,
+    ModalContent,
+    ModalHeader,
+    ModalCloseButton,
+    ModalBody,
+    ModalFooter,
+} from '@chakra-ui/react'
+import { useState } from 'react'
+
+export type SignInCardProps = {}
+
+const SignInCard = ({}: SignInCardProps) => {
+    const { signMessageAsync } = useSignMessage()
+    const { address, isConnected } = useAccount()
+    const { connect } = useConnect({
+        connector: new InjectedConnector(),
+    })
```

Comment:
nit: We probably can split this into client and server callable components right? since useSession() is available on next server, but im assuming the WAGMI injector needs the client browser. That way, we reduce the size of the client code

Reply:
thats true